### PR TITLE
Gutenberg: include @wordpress/data package

### PR DESCRIPTION
**Pull request by vindl**
Part of https://github.com/Automattic/wp-calypso/issues/26180

Set up a Calypso test branch for `@wordpress/data` package and add some basic tests for it.

### Testing instructions

- `npm run test-client client/gutenberg/editor/test/index.js`
- Calypso smoke test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,8 @@
     "@babel/preset-react": "7.0.0-beta.44",
     "@babel/preset-stage-2": "7.0.0-beta.44",
     "@babel/runtime": "7.0.0-beta.44",
+    "@wordpress/data": "1.0.0",
+    "ajv": "6.5.2",
```
Comment (blowery) [inline package.json]: why the ajv dep?

Comment (vindl) [reply package.json]: I was running into this during installation:

`npm WARN ajv-keywords@3.2.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.`

Haven't checked if it's really necessary yet.

Comment (blowery) [reply package.json]: ah ok. i think that's a npm issue. we see it on master too.

Comment (blowery) [reply package.json]: https://github.com/webpack-contrib/schema-utils/issues/23#issuecomment-367322144

Comment (vindl) [reply package.json]: Thanks for the additional context! As expected, removing it didn't cause any issues on this branch.

**Comment by gwwar** [comment]
💖looks promising. Let's move over testing registerStore and dispatch into a unit test. Since Gutenberg is developing so quickly this will help let us know if anything has changed with future package updates. We can maybe remove this later once things stabilize a bit.

After we verify that we haven't added bloat to other bundles (besides the gutenberg section) http://iscalypsofastyet.com/branch?branch=try/wordpress-data I think we can land this.

**Comment by gwwar** [comment]
CI errors are from having an empty test file:
```
 FAIL  client/gutenberg/editor/test/index.js
  ● Test suite failed to run

    Your test suite must contain at least one test.

      at node_modules/jest/node_modules/jest-cli/build/test_scheduler.js:256:22

Summary of all failing tests
 FAIL  client/gutenberg/editor/test/index.js
  ● Test suite failed to run

    Your test suite must contain at least one test.

      at node_modules/jest/node_modules/jest-cli/build/test_scheduler.js:256:22
```

**Comment by vindl** [comment]
> Let's move over testing registerStore and dispatch into a unit test.

@gwwar moved these and a couple of other `@wordpress/data` exported functions to a unit test. This should also resolve the CI errors.

File: client/gutenberg/editor/test/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
+/**
+ * @format
+ * @jest-environment jsdom
```
[the remark this answers is not in the document]
Comment (vindl) [reply client/gutenberg/editor/test/index.js]: @gwwar Without it the test will produce this error:

> ReferenceError: window is not defined
>      at registerReducer (node_modules/@wordpress/data/build/registry.js:215:5)
> ...

Quoting the testing docs:

> It's very possible that your tests will assume the existence of a browser environment to work properly. The test runner we use, Jest, uses the browser-like environment jsdom. We default to a node-like environment to make tests faster. If some tests require another environment, you can add /** @jest-environment jsdom */ docblock. Check this Jest doc to learn more.

(Reference: https://github.com/Automattic/wp-calypso/blob/a9150ed7ded1f97235bd4e069639ea9b1ab3d6e7/docs/testing/component-tests.md#set-up-a-test-environment).

It looks like the `registerReducer` depends on `window`, so I had to include this to make tests happy.

**Pull request merged**